python/ml4ir/base/model/architectures/flexible_dense_model.py

```diff
@@ -0,0 +1,253 @@
+from __future__ import annotations
```

Comment:
@arvindsrikantan Worth renaming this module name too. `flexible_dense_model` is in some ways all encompassing.

Reply:
That is the idea though, haha